WooCommerce/Classes/Extensions/UINavigationController+Woo.swift

```diff
@@ -57,30 +57,20 @@ extension UIViewController: UINavigationBarBackButtonHandler {
 }
 
 extension UINavigationController: UINavigationBarDelegate {
-
-    // This delegate method is not called on the simulator or device running iOS 13.4 from Xcode.
-    // You need to use a release build.
-    public func navigationBar(_ navigationBar: UINavigationBar, shouldPop item: UINavigationItem) -> Bool {
-        guard let items = navigationBar.items else {
-            return false
-        }
-
-        if viewControllers.count < items.count {
-            return true
-        }
-
+    public func checkIfNavigationBarShouldPop(_ navigationBar: UINavigationBar, item: UINavigationItem) -> Bool {
         var shouldPop = true
 
         if let vc = topViewController {
             shouldPop = vc.shouldPopOnBackButton()
         }
 
-        if shouldPop {
-            DispatchQueue.main.async {
-                self.popViewController(animated: true)
-            }
-        }
-        return false
```

Comment:
Since this delegate method checks if bar should pop I figured it is best to just return if it should pop or not and let the system do the popping itself.